### PR TITLE
ConnectionInterface typehint

### PR DESCRIPTION
**Pull request by HavokInspiration**
Typehint should be against the `ConnectionInterface` and not a `Connection` object.

### CODE REVIEW

File: src/ORM/Table.php [hunk 1]
```diff
@@ -389,10 +389,10 @@ public function registryAlias($registryAlias = null)
     /**
      * Returns the connection instance or sets a new one
      *
-     * @param \Cake\Database\Connection|null $conn The new connection instance
-     * @return \Cake\Database\Connection
+     * @param \Cake\Datasource\ConnectionInterface|null $conn The new connection instance
+     * @return \Cake\Datasource\ConnectionInterface
      */
-    public function connection(Connection $conn = null)
+    public function connection(ConnectionInterface $conn = null)
```
Comment (dereuromark) [inline src/ORM/Table.php]: We would need to communicate that as it will break userland typehints in overwriting/extending classes, will it not?

Comment (HavokInspiration) [inline src/ORM/Table.php]: It's a good point yes.

Comment (dereuromark) [inline src/ORM/Table.php]: I figure almost no one would have overwritten this method, though :)

Comment (markstory) [inline src/ORM/Table.php]: I still think this should go into 3.1 as that's when the interface was added and this was omitted.

**Pull request merged**